packages/frontend/app/services/channels/vote-verifier.ts

```diff
@@ -0,0 +1,48 @@
+import Service, { inject as service } from '@ember/service';
+import VoteChain from 'emberclear/models/vote-chain';
+import CryptoConnector from '../workers/crypto';
+import WorkersService from '../workers';
+import { generateSortedVote } from './-utils/vote-sorter';
+import { equalsUint8Array } from 'emberclear/utils/uint8array-equality';
+
+export default class VoteVerifier extends Service {
+  @service workers!: WorkersService;
+  crypto?: CryptoConnector;
+
+  async verify(voteToVerify: VoteChain): Promise<boolean> {
+    this.connectCrypto();
+
+    let voteToVerifyActual: Uint8Array = generateSortedVote(voteToVerify);
+    let voteToVerifyActualHash: Uint8Array = await this.crypto!.hash(voteToVerifyActual);
```

Comment:
Adding the exclamation point here isn't quite correct, because 'this.crypto' might be undefined still, like if connect crypto silently fails.

You'll want to have a guard condition above here and drop the exclamation ;)

Reply:
Alternatively

The method could be dropped and the property initialized with the connector.

Reply:
Would there be a way to set up some sort of retry thing to prevent this. Ultimately since this is a verifier, I wouldn't really want to have either boolean if the method were to fail due to not being able to connect to the worker.

Reply:
Yeah, but let's not worry about that scope creep yet. I need to figure out a global retry system for when anything goes wrong so the user can retry manually, or we automatically retry with a some backoff timer or something.

For now, let's assume verification fails if there is crypto worker

Reply:
Resolved in 0e75e91